### PR TITLE
Fixing diag trace file bug

### PR DESCRIPTION
**Pull request by prawalagarwal**
The VsTest v2 task today doesn't handle the diag file when system.debug is set to true. The file is created but we do not upload it to the build logs. This change handles this issue, plus the issue where search location in the task if not a directory, we don't get any meaningful message in the logs.

1) We are making search folder field compulsory in the task.
2) The validation if the search folder is a valid field is moved to taskinputparser.
4) CI publish method is moved to Helper.ts

### CODE REVIEW

File: Tasks/VsTest/vstest.ts [hunk 1]
```diff
@@ -485,6 +490,7 @@ function cleanFiles(responseFile: string, listFile: string, testCaseFilterFile:
 
 function deleteVstestDiagFile(): void {
```
Comment (kaadhina) [inline Tasks/VsTest/vstest.ts]: can we rename this to say UploadAndDelete

File: Tasks/VsTest/vstest.ts [hunk 2]
```diff
@@ -485,6 +490,7 @@ function cleanFiles(responseFile: string, listFile: string, testCaseFilterFile:
...
     if (vstestConfig && vstestConfig.vstestDiagFile && pathExistsAsFile(vstestConfig.vstestDiagFile)) {
+        console.log('##vso[task.uploadfile]'+ vstestConfig.vstestDiagFile);   
         tl.debug('Deleting vstest diag file ' + vstestConfig.vstestDiagFile);
         tl.rmRF(vstestConfig.vstestDiagFile);
```
Comment (kaadhina) [inline Tasks/VsTest/vstest.ts]: Is it possible the file gets deleted before it gets uploaded. Since upload is async

Comment (kaadhina) [reply Tasks/VsTest/vstest.ts]: Actually I  think _temp directory gets auto cleaned after the job completes. This rmRf most likely is redundant and can be removed. Please check.

File: Tasks/VsTest/vstest.ts (hunk 1 above)
Comment (prawalagarwal) [reply Tasks/VsTest/vstest.ts]: Yes

File: Tasks/VsTest/vstest.ts (hunk 2 above)
Comment (prawalagarwal) [reply Tasks/VsTest/vstest.ts]: Yes this is a possibility, we should remove the deletion of diag file

File: Tasks/VsTest/vstest.ts [hunk 3]
```diff
@@ -91,8 +92,13 @@ function getTestAssemblies(): string[] {
         tl.debug('Search directory empty, defaulting to ' + vstestConfig.testDropLocation);
     }
 
-    tl.debug('Searching for test assemblies in: ' + vstestConfig.testDropLocation);
-    return tl.findMatch(vstestConfig.testDropLocation, vstestConfig.sourceFilter);
+    if (pathExistsAsDirectory(vstestConfig.testDropLocation)) {
```
Comment (RanjanarMS) [inline Tasks/VsTest/vstest.ts]: Can this check be added in taskInputParser only?

Comment (RanjanarMS) [reply Tasks/VsTest/vstest.ts]: Is it a warning or error?

File: Tasks/VsTest/vstest.ts [hunk 4]
```diff
@@ -490,10 +496,9 @@ function cleanFiles(responseFile: string, listFile: string, testCaseFilterFile:
     tl.rmRF(testCaseFilterOutput);
 }
 
-function deleteVstestDiagFile(): void {
+function uploadVstestDiagFile(): void {
     if (vstestConfig && vstestConfig.vstestDiagFile && pathExistsAsFile(vstestConfig.vstestDiagFile)) {
-        tl.debug('Deleting vstest diag file ' + vstestConfig.vstestDiagFile);
-        tl.rmRF(vstestConfig.vstestDiagFile);
+        console.log('##vso[task.uploadfile]'+ vstestConfig.vstestDiagFile);
```
Comment (RanjanarMS) [inline Tasks/VsTest/vstest.ts]: is there any size restriction for the vstestDiagFile?

Comment (RanjanarMS) [reply Tasks/VsTest/vstest.ts]: Can we put a CI event for whether the diag file is created and its size?

File: Tasks/VsTest/vstest.ts (hunk 3 above)
Comment (prawalagarwal) [reply Tasks/VsTest/vstest.ts]: This check doesn't prints anything by itself. We will print a warning in the else clause.

File: Tasks/VsTest/vstest.ts (hunk 4 above)
Comment (cltshivash) [reply Tasks/VsTest/vstest.ts]: log message should fine. Ranjana ?

**Pull request merged**